tests/test_extensions.py

```diff
@@ -339,6 +346,35 @@ def test_anonymize_with_anonymizer(self):
         assert recipe.all()[0].age == 10
         assert recipe.stats.rows == 2
 
+    def test_anonymize_with_faker_anonymizer(self):
+        """ Anonymize requires ingredients to have an anonymizer """
+        recipe = self.recipe().metrics('age').dimensions('firstanon').order_by(
+            'firstanon'
+        ).anonymize(False)
+        assert recipe.to_sql() == """SELECT foo.first AS firstanon,
+       sum(foo.age) AS age
+FROM foo
+GROUP BY foo.first
+ORDER BY foo.first"""
+        assert recipe.all()[0].firstanon == 'hi'
+        assert recipe.all()[0].firstanon_id == 'hi'
+        assert recipe.all()[0].age == 15
+        assert recipe.stats.rows == 1
+
+        recipe = self.recipe().metrics('age').dimensions('firstanon').order_by(
+            'firstanon'
+        ).anonymize(True)
+        assert recipe.to_sql() == """SELECT foo.first AS firstanon_raw,
+       sum(foo.age) AS age
+FROM foo
+GROUP BY foo.first
+ORDER BY foo.first"""
+        assert recipe.all()[0].firstanon != 'hi'
```

Comment:
I think we can do better than just asserting that it's not `'hi'` -- we should be able to invoke the FakerFormatter (or faker directly) with the right arguments to make sure that we're getting the expected result.